packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -6,6 +6,7 @@ import 'dart:math' as math;
 
 import 'package:flutter/widgets.dart';
 
+import '../foundation/memory_allocations.dart';
```

Comment:
```suggestion
import 'package:flutter/foundation.dart';
```

Reply:
Hm, I thought we prefer relative import: https://dart.dev/tools/linter-rules/prefer_relative_imports
Absolute import creates circular reference here:  memory_allocations.dart > foundation.dart -> change_notifier.dart -> memory_allocations.dart, that makes it harder to detect layering violations.

Reply:
But, I see, there is no consistency in Flutter Framework here. I see both relative and absolute imports.

Reply:
I personally prefer relative, so I will leave it as is, but may be we need a team level agreement and lint here :)

Reply:
For cross library import we use absolute imports. Only in library import we use relative import. We need to change this back, can you create a new pr for this?

Reply:
https://github.com/flutter/flutter/pull/135337